json-endpoints/class.wpcom-json-api-post-endpoint.php

```diff
@@ -407,7 +407,7 @@ function get_post_by( $field, $post_id, $context = 'display' ) {
 				break;
 			case 'attachments':
 				$response[$key] = array();
-				$_attachments = get_posts( array( 'post_parent' => $post->ID, 'post_status' => 'inherit', 'post_type' => 'attachment' ) );
+				$_attachments = get_posts( array( 'post_parent' => $post->ID, 'post_status' => 'inherit', 'post_type' => 'attachment', 'posts_per_page' => -1 ) );
```

Comment:
Does this conflict with [**No `limit` queries**](http://vip.wordpress.com/documentation/code-review-what-we-look-for/#no-limit-queries)? I suppose it would be unlikely for there to be more than 100 attachments on a given post, though.